astropy/utils/data.py

```diff
@@ -871,10 +871,72 @@ def compute_hash(localfn):
     return h.hexdigest()
 
 
+def get_pkg_data_path(*path, package=None):
+    """Get path from source-included data directories.
+
+    Parameters
+    ----------
+    *path : str
+        Name/location of the desired data file/directory.
+        May be a tuple of strings for ``os.path`` joining.
+
+    package : str or `None`, optional, keyword only
+        If specified, look for a file relative to the given package, rather
+        than the calling module's package.
+
+    Returns
+    -------
+    path : str
+        Name/location of the desired data file/directory.
+
+    Raises
+    ------
+    ImportError
+        Given package or module is not importable.
+    """
+    if package is None:
+        module = find_current_module(1, finddiff=['astropy.utils.data', 'contextlib'])
+        if module is None:
+            # not called from inside an astropy package.  So just pass name
+            # through
+            return os.path.join(*path)
+
+        if not hasattr(module, '__package__') or not module.__package__:
+            # The __package__ attribute may be missing or set to None; see
+            # PEP-366, also astropy issue #1256
+            if '.' in module.__name__:
+                package = module.__name__.rpartition('.')[0]
+            else:
+                package = module.__name__
+        else:
+            package = module.__package__
+    else:
+        module = resolve_name(package)
+
+    rootpkgname = package.partition('.')[0]
+
+    rootpkg = resolve_name(rootpkgname)
+
+    module_path = os.path.dirname(module.__file__)
+    path = os.path.join(module_path, *path)
+
+    root_dir = os.path.dirname(rootpkg.__file__)
+    if not _is_inside(path, root_dir):
+        raise RuntimeError(f"attempted to get a local data file outside "
+                           f"of the {rootpkgname} tree.")
+
+    return path
```

Comment:
I'm very confused as to why this is duplicating the entire implementation of `_find_pkg_data_path`.  Unless I'm missing something I think all this needs to be is:

```python
def get_pkg_data_path(*path, package=None):
    """
    ... docstring ...
    """
    return _find_pkg_data_path(os.path.join(path), package=package)

Reply:
Would it be better to do the reverse, have ``_find_pkg_data_path`` call ``get_pkg_data_path``? As get_pkg_data_path is publicly scoped it is guaranteed to be more stable and makes clear that _find_pkg_data_path might be deprecated at some future date.

Reply:
You could do it that way, but it would actually be more trouble I think to split the path than to join it (splitting is actually more error-prone).  I'd say leave the implementation in `_find_pkg_data_path`.  There's currently no plans to remove it.

Reply:
I can do that, but there shouldn't be any need to split the path. If it comes as a single string, ``get_pkg_data_path`` assumes it's already joined.

Reply:
Ok, that makes sense.  As a matter of consistency I would make the public interface depend on the private interface, not the other way around.  But if you want to do it this way I'm fine with that.  Then maybe go ahead and add a `PendingDeprecation` warning to `_find_pkg_data_path`.

Reply:
You can go ahead and add

```python
from astropy.utils.decorators import deprecated

@deprecated('4.3', alternative='Use astropy.utils.data.get_pkg_data_path', pending=True)
def _find_pkg_data_path(data_name, package=None):
    ....
```